horovod/runner/elastic/discovery.py

```diff
@@ -15,20 +15,33 @@
 
 import io
 import logging
+import random
 import threading
-
+import time
 from collections import defaultdict
 
 from horovod.runner.common.util import safe_shell_exec
 from horovod.runner.elastic.worker import HostUpdateResult
-
-
+# Maximum time hosts are in cooldown period
+COOLDOWN_UPPER_LIMIT_SECONDS = 5 * 60
+# Minimum time hosts are in cooldown
+COOLDOWN_LOWER_LIMIT_SECONDS = 10
+# Unit of increasing cooldown for repeat failure
+COOLDOWN_RANGE_SECONDS = [COOLDOWN_LOWER_LIMIT_SECONDS, COOLDOWN_UPPER_LIMIT_SECONDS]
 class HostState(object):
-    def __init__(self):
+
+    def __init__(self, cooldown_range=None):
         self._event = threading.Event()
 
-        # TODO(travis): blacklisted hosts should have a timeout period that increases with each failure
         self._blacklisted = False
+        self._blacklist_count = 0
+        if cooldown_range:
+            self._cooldown_lower_limit = cooldown_range[0]
```

Comment:
Does this require some sanity check on the values in `cooldown_range`?

Reply:
Added